npm/cmd/start.go

```diff
@@ -115,8 +117,20 @@ func start(config npmconfig.Config, flags npmconfig.Flags) error {
 	factor := rand.Float64() + 1 //nolint
 	resyncPeriod := time.Duration(float64(minResyncPeriod.Nanoseconds()) * factor)
 	klog.Infof("Resync period for NPM pod is set to %d.", int(resyncPeriod/time.Minute))
-	factory := informers.NewSharedInformerFactory(clientset, resyncPeriod)
 
+	factory := informers.NewSharedInformerFactory(clientset, resyncPeriod)
+	// npm-lite -> daemon set will listen to pods only in its own node
+	if config.Toggles.EnableNPMLite {
+		nodeName := os.Getenv("NODE_NAME")
+		factory = informers.NewSharedInformerFactoryWithOptions(
+			clientset,
+			resyncPeriod,
+			informers.WithTweakListOptions(func(options *metav1.ListOptions) {
+				// Use field selector to filter pods based on their assigned node
+				options.FieldSelector = "spec.nodeName=" + nodeName
```

Comment:
Is there a way to only modify the pod informer? The current change might not impact ns and netpol informers anyways, but perhaps it's safer to scope changes to only the pod informer?

Reply:
I'll look more into it and check

Reply:
@huntergregory, 
I changed the code so that the informer change is only scoped to pod controller. Please let me know if that looks better

Reply:
Looks good! Thanks